NumLib/DOF/LocalToGlobalIndexMap.cpp

```diff
@@ -242,6 +276,34 @@ LocalToGlobalIndexMap* LocalToGlobalIndexMap::deriveBoundaryConstrainedMap(
                                      std::move(mesh_component_map));
 }
 
+LocalToGlobalIndexMap* LocalToGlobalIndexMap::deriveBoundaryConstrainedMap(
+    int const variable_id,
+    std::vector<int> const& component_ids,
+    std::unique_ptr<MeshLib::MeshSubsets>&& mesh_subsets,
+    std::vector<MeshLib::Element*> const& elements) const
+{
+    DBUG("Construct reduced local to global index map.");
+    // Create a subset of the current mesh component map.
+    std::vector<std::size_t> global_component_ids;
+
+    for (auto component_id : component_ids)
+        global_component_ids.push_back(
+            getGlobalComponent(variable_id, component_id));
+
+    auto mesh_component_map =
+        _mesh_component_map.getSubset(global_component_ids, *mesh_subsets);
+
+    // Create copies of the mesh_subsets for each of the global components.
+    std::vector<std::unique_ptr<MeshLib::MeshSubsets>> all_mesh_subsets;
+    for (std::size_t i = 0; i < global_component_ids.size(); ++i)
+        all_mesh_subsets.emplace_back(
+            new MeshLib::MeshSubsets{*mesh_subsets});
```

Comment:
Above you `std::move(mesh_subsets)`, here you don't. So IMO passing `mesh_subsets` as &&-reference is not needed here.